index.js

```diff
@@ -1,55 +1,94 @@
 const WebSocket = require('ws');
 const request = require('request');
 const getParser = require('@streammedev/parse-message');
+const program = require('commander');
 
-const botKey = 'ADD THE BOT KEY';
-const botSecret = 'ADD THE BOT SECRET';
+program
+	.version('0.0.1')
+	.option('-k, --key <key>', 'bot key')
+	.option('-s, --secret <secret>', 'bot secret')
+	.option('-i, --id <id>', 'your public id')
+	.parse(process.argv);
+	
+var botKey = program.key;
+var botSecret = program.secret;
+var publicId = program.id;
+
+// Used to check if id and bot key are valid
+const uuidRegEx = /^[a-fA-F0-9]{8}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{12}$/;
+
+if (typeof botKey === 'undefined' || !uuidRegEx.test(botKey)) {
+	if (process.env.botKey) {
+		botKey = process.env.botKey;
+	} else {
+		console.error('invalid key given');
+		process.exit(1);
+	}
+}
+if (typeof botSecret === 'undefined') {
+	if (process.env.botSecret) {
+		botSecret = process.env.botSecret;
+	} else {
+		console.error('no secret given');
+		process.exit(1);
+	}
+}
+if (typeof publicId === 'undefined' || !uuidRegEx.test(publicId)) {
+	if (process.env.publicId) {
+		publicId = process.env.publicId;
+	} else {
+		console.error('invalid id given');
+		process.exit(1);
+	}
+}
 
-const publicId = 'ADD THE PUBLIC ID OF THE CHANNEL';
 const roomId = `user:${publicId}:web`;
 
+const baseWaitMS = 500;
 const secondsInMS = 1000;
 const minutesInMS = secondsInMS * 60;
 const hoursInMS = minutesInMS * 60;
 const daysInMS = hoursInMS * 24;
 const monthsInMS = daysInMS * 30;
 const yearsInMS = daysInMS * 365;
 
-function botAuth (cb) {
-	request({
-		method: 'POST',
-		url: 'https://www.stream.me/api-auth/v1/login-bot',
-		body: {
-			key: botKey,
-			secret: botSecret
-		},
-		json: true
-	}, function (err, res, body) {
-		if (err) {
-			console.log(err);
-			return;
-		}
-		if (res.statusCode !== 200) {
-			console.log('Could not authenticate bot: Did not receive 200 http status, instead was: ', resp.statusCode, ' body: ', body);
-			return cb(new Error('Could not authenticate: non 200 response'));
-		}
-		const botToken = body.access_token;
-		cb(err, botToken);
-	});
-}
+const botAuth = retry(function (cb) {
+		request({
+			method: 'POST',
+			url: 'https://www.stream.me/api-auth/v1/login-bot',
+			body: {
+				key: botKey,
+				secret: botSecret
+			},
+			json: true
+		}, function (err, res, body) {
+			if (err) {
+				console.log(err);
```

Comment:
use pino instead of console.log

https://www.npmjs.com/package/pino

it will include things that are required for logs - timestamps, levels, etc